### PR TITLE
caddy: update to 2.3.0

### PR DESCRIPTION
**Pull request by gpanders**
#### Description

Update caddy to 2.3.0.

Caddy 2 has a bit of an unorthodox method of adding the version information into the final binary. Where many Go programs use the ldflags trick, Caddy gets its version information from the debug metadata that Go itself bakes into the Caddy module. This means that in order to get the correct version string from `caddy version`, you have to create a new Go module with the version of Caddy you want as a dependency ([see the README](https://github.com/caddyserver/caddy#with-version-information-andor-plugins) for more information). This means this port **must** be built in module mode, so for now the `build.env-delete GO111MODULE=off GOPROXY=off` line is necessary.

Note that if/when #10381 is accepted, we may be able to use that approach to add a `go.vendors` line into this port. But until then, it's not possible (as far as I can tell, I spent a few hours trying to get this to work).

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7 19H524
Xcode 12.4 12D4e

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [X] checked your Portfile with `port lint`?
- [X] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

File: www/caddy/Portfile [hunk 1]
```diff
@@ -3,32 +3,34 @@
...
+# Caddy uses its own tool (xcaddy) to build the caddy executable with version
+# information baked in
+go.setup            github.com/caddyserver/xcaddy 0.1.8 v
+
+name                caddy
+version             2.3.0
+revision            0
 categories          www
 platforms           darwin
 supported_archs     i386 x86_64
```
Comment (herbygillot) [inline www/caddy/Portfile:15]: Let's remove `supported_archs` so that the golang portgroup does the right thing automatically on whichever architecture.

Comment (gpanders) [reply www/caddy/Portfile]: Done

**Comment by herbygillot** [comment]
Thanks @gpanders.  @mohd-akram, any comments here?

**Comment by mohd-akram** [comment]
LGTM, thanks @gpanders, and @herbygillot for reviewing.

**Review by mohd-akram — APPROVED** [review]

**Pull request merged**